gamemode/cl_bannedmodels.lua

```diff
@@ -1,147 +1,186 @@
-GM.BannedModels = {}
+-- This file provides the functionality for accessing the banned models
+-- menu. It is used locally to determine if a model is viable (aka
+-- whether or not it gets an outline when moused over).
+
+
+GM.BannedModels = {} -- This is used as a hash table where the key is the model string and the value is true.
+local menu
+
 
 function GM:IsModelBanned(model)
-	return table.HasValue(self.BannedModels, model)
+	return self.BannedModels[model] == true
 end
 
+
 function GM:AddBannedModel(model)
-	table.insert(self.BannedModels, model)
+	if self.BannedModels[model] == true then return end -- Prevent duplicates.
+
+	self.BannedModels[model] = true
+	menu.AddModel(model)
 end
 
+
 function GM:RemoveBannedModel(model)
-	table.RemoveByValue(self.BannedModels, model)
-end
+	if self.BannedModels[model] != true then return end -- Check if exists before trying to remove.
 
-function GM:GetBannedModels()
-	return self.BannedModels
+	self.BannedModels[model] = nil
+	menu.RemoveModel(model)
 end
 
-net.Receive("ph_bannedmodels", function (len)
+
+net.Receive("ph_bannedmodels_getall", function (len)
 	GAMEMODE.BannedModels = {}
 
-	while true do
-		local k = net.ReadUInt(16)
-		if k == 0 then
-			break
-		end
-		local v = net.ReadString()
-		GAMEMODE.BannedModels[k] = v
+	local banned_model = net.ReadString()
+	while banned_model != "" do
+		GAMEMODE:AddBannedModel(banned_model)
+		banned_model = net.ReadString()
 	end
 end)
 
-local menu
+
+net.Receive("ph_bannedmodels_add", function (len)
+	local v = net.ReadString()
+	GAMEMODE:AddBannedModel(v)
+end)
+
+
+net.Receive("ph_bannedmodels_remove", function (len)
+	local v = net.ReadString()
+	GAMEMODE:RemoveBannedModel(v)
+end)
+
+
 concommand.Add("ph_bannedmodels_menu", function (client)
-	if !client:IsSuperAdmin() then
-		if IsValid(menu) then
-			menu:Remove()
-		end
-		return
-	end
-	if IsValid(menu) then
-		menu:SetVisible(true)
-		net.Start("ph_bannedmodels")
-		net.SendToServer()
-		return
-	end
+	menu:SetVisible(true)
+end)
 
-	-- GAMEMODE.BannedModels = {"asd/models.ad"}
 
+-- This is all the code to create the banned models menu.
+-- TODO: I don't like adding this to GM. I feel like there should absolutely be a better solution.
+function GM:CreateBannedModelsMenu()
+	-- The main window that will contain all of the functionality for display, adding, and
+	-- removing banned models.
 	menu = vgui.Create("DFrame")
+	menu:SetPos(200, 200)
```

Comment:
What is this SetPos for? Doesn't the `Center()` below handle this?

Reply:
Definitely not needed, I have removed it.